examples/profile.js

```diff
@@ -92,22 +181,20 @@ app.MainController = function($http, $scope) {
     }
   };
 
-
-  // Using closures for hoverCallback and outCallback since
-  // wrapping in angular.bind leads to a closure error.
-  // See PR https://github.com/google/closure-compiler/pull/867
-  var that = this;
-
   /**
    * @param {Object} point
    */
   var hoverCallback = function(point) {
     // An item in the list of points given to the profile.
     that['point'] = point;
+    that.snappedPoint_.setGeometry(new ol.geom.Point([point.x, point.y]));
+    $scope.$apply();
```

Comment:
I just tried:
https://github.com/pgiraud/ngeo/commit/793dec6f94f0eaed0b75d228557b794525031132

But I then get a "too much recursion" error.
Did you have something similar in mind?